src/core/etl/tests/Flow/ETL/Tests/Integration/Row/Reference/Expression/AddJsonTest.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\ETL\Tests\Integration\Row\Reference\Expression;
+
+use function Flow\ETL\DSL\lit;
+use function Flow\ETL\DSL\ref;
+use Flow\ETL\DSL\Entry;
+use Flow\ETL\DSL\From;
+use Flow\ETL\DSL\To;
+use Flow\ETL\Flow;
+use Flow\ETL\Memory\ArrayMemory;
+use Flow\ETL\Row;
+use Flow\ETL\Rows;
+use PHPUnit\Framework\TestCase;
+
+final class AddJsonTest extends TestCase
+{
+    public function test_add_json_into_existing_reference() : void
+    {
+        (new Flow())
+            ->read(
+                From::rows(
+                    new Rows(
+                        Row::with(Entry::int('id', 1), Entry::array('array', ['a' => 1, 'b' => 2, 'c' => 3])),
+                    )
+                )
+            )
```

Comment:
for a simple scenarios like this where types of entries can be easily determined by NativeEntryFactory it would be more readable to use 

```php
From::array([
    [['id' => 1, 'array' => ['a' => 1...]]
])
```

Reply:
It's not exactly a 1:1 replacement for manual rows building, but it's true it's a more production-like example.

To cover this case properly it requires a few more calls:
```php
        (new Flow())
            ->read(
                From::array(
                    [['id' => 1, 'array' => ['a' => 1, 'b' => 2, 'c' => 3]]],
                )
            )
            ->withEntry('row', array_unpack(ref('row')))
            ->renameAll('row.', '')
            ->drop('row')